pyne/mcnp.py

```diff
@@ -1204,8 +1204,13 @@ def read_variable_ids(self):
         variable_ids = dict()
 
         if self.eightbytes:
-            variable_info = self.read_next(
-                "qqqqqqqqqqqiiiiiiiii", 124, raw_format=True)
+            mcnp_version=self.mcnp_version_info.split()[1]
+            if mcnp_version in ["6","6.mpi"]:
+                variable_info = self.read_next(
+                    "iqqqqqqqqqqiiiiiiiii", 120, raw_format=True)
+            else: # Not sure about MCNPX 
+                variable_info = self.read_next(
+                    "qqqqqqqqqqqiiiiiiiii", 124, raw_format=True)
```

Comment:
I am not sure I understand this change.
Could you add a comment to explain briefly why `mcnp6` is not read the same way as the other `mcnp` version ?
(i.e. why `iqqqqqqqqqqiiiiiiiii, 120` in one case and `qqqqqqqqqqqiiiiiiiii, 124` in the other)

Reply:
Well, it is  the thing here is that the variable MCNP uses in `ptrak()` for the first value in that register is explicitly declared as `i8knd` in MCNP5, and as `integer` in MCNP6, thus the change in format and length. 

However, on second thought, it is possible to compile MCNP6 with 8 byte integers by default, IIRC, so I may have to try different compile options and test again...

Reply:
The 8 bit integers in MCNP6 is quite a can of worms, as even the first registry is no longer what we expect. I am adding an exception to handle the situation and calling it a day. If there is interest in it (i.e: users actually using such compilation options), support can always be added later.